packages/utils/src/scheduleAtApproximateInterval.ts

```diff
@@ -0,0 +1,32 @@
+import { scheduleAtInterval } from './scheduleAtInterval'
+import { wait } from './wait'
+
+/**
+ * @param {number} approximateIntervalInMs - approximate time (in milliseconds) to wait after a task is completed
+ * @param {number} driftMultiplier how much the wait time can vary: e.g. if the interval is 60 minutes and the drift is 0.1,
+ * the delay between invocations will range from 54 to 66 minutes
+ */
+export const scheduleAtApproximateInterval = async (
+    task: () => Promise<void>,
+    approximateIntervalInMs: number,
+    driftMultiplier: number,
+    executeAtStart: boolean,
+    abortSignal: AbortSignal
+): Promise<void> => {
+    if (abortSignal?.aborted) {
```

Comment:
```suggestion
    if (abortSignal.aborted) {
```
I guess abortSignal cannot be undefined.